app/models/poll.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+class Poll < ActiveRecord::Base
+  has_many :options, dependent: :destroy
+
+  validates :title, presence: true
+
+  after_touch :ensure_status_is_correct
+
+  def make_ready
```

Comment:
And about the enums — we have a really great experience with Postgres enumerable datatypes as a ActiveRecord::Enum backend.
https://www.sitepoint.com/enumerated-types-with-activerecord-and-postgresql/